go/types/list.go

```diff
@@ -162,24 +162,30 @@ func (l List) RemoveAt(idx uint64) List {
 type listIterFunc func(v Value, index uint64) (stop bool)
 
 func (l List) Iter(f listIterFunc) {
+	l.IterFrom(0, f)
+}
+
+type listIterAllFunc func(v Value, index uint64)
+
+func (l List) IterAll(f listIterAllFunc) {
 	idx := uint64(0)
 	cur := newCursorAtIndex(l.seq, idx)
 	cur.iter(func(v interface{}) bool {
-		if f(v.(Value), uint64(idx)) {
-			return true
-		}
+		f(v.(Value), uint64(idx))
 		idx++
 		return false
 	})
 }
 
-type listIterAllFunc func(v Value, index uint64)
-
-func (l List) IterAll(f listIterAllFunc) {
-	idx := uint64(0)
+func (l List) IterFrom(idx uint64, f listIterFunc) {
```

Comment:
In one case I went through those contortions, definitely. It was really only because I wanted to co-iterate across two lists at the same time. In the other case, the callback model was just fine, so I don't feel like this code gives a clear mandate one way or the other :-/

Do you have other potential use cases that might break the tie?